python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -1644,6 +1685,16 @@ def serialize(self) -> str:
                     for window in self.included_time_windows
                 ],
                 "num_partitions": self._num_partitions,
+                "time_partitions_def": serialize_value(
+                    TimeWindowPartitionsDefinition(
```

Comment:
Is there a way to automatically cast an instance of a subclass into its superclass?

Unfortunately subclasses of `TimeWindowPartitionsDefinition` can't be automatically serialized (I think we would need to add the whitelist_for_serdes tag on them too)

Reply:
If we only serialized partitions defs from external asset graphs we could probably remove these lines